### DISCUSSION TITLE
Postgres SSL not working

### DISCUSSION THREAD

**Discussion by krunde-amfam** [prisma/prisma#18794] [Q&A] [answered]
Hello,

When I switched from a Linux laptop to a MBP I can no longer connect to Postgres DBs using SSL. I have tried regenerating the .p12 files using the built in openssl, openssl 3, openssl 1.1 (see exact versions below). Here is the error I get:
```
Error: P1011

Error opening a TLS connection: MAC verification failed during PKCS12 import (wrong password?)
```

I'm at a loss for what else to try. I can connect to the DB using psql and the cert and key file. So I know those are valid. From what I can tell the .p12 file is valide, but since no other Postgres tool seem to support that I'm not sure how to validate that.

```
% openssl version
LibreSSL 3.3.6
% /opt/homebrew/opt/openssl@3/bin/openssl version
OpenSSL 3.1.0 14 Mar 2023 (Library: OpenSSL 3.1.0 14 Mar 2023)
% /opt/homebrew/opt/openssl@1.1/bin/openssl version
OpenSSL 1.1.1t  7 Feb 2023
```

Thank You,
Kevin

**Comment by ludralph** [comment]
Hi Kevin

It appears you are encountering an issue with SSL certificate verification when connecting to Postgres databases on your MacBook Pro. The error message you're seeing suggests that the issue may be related to the password used to encrypt the private key in the PKCS12 file. You might want to verify that the password used to encrypt the private key in the PKCS12 file is correct. You can do this by attempting to import the PKCS12 file into another application, such as Keychain Access, and entering the same password. If the password is incorrect, you may need to regenerate the PKCS12 file with a new password.

**Reply by krunde-amfam** [reply]
Thank you for the quick reply. The probably was that I was using a blank password. Other things that can check a .p12 file work with that fine, but Prisma does not. Not sure why that change between Linux and Mac OS. Thank you for making me rethink things!

Thank you,
Kevin

**Reply by ludralph** [reply, marked as answer]
I'm glad I could help. Can you please mark this discussion as answered if you do not have further questions?
